### PR TITLE
Add run_js to pyodide Python package

### PR DESCRIPTION
**Pull request by hoodmane**
### Description

Add a `run_js` API that evals JavaScript code and returns the result as a `JsProxy`.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

**Comment by ryanking13** [comment]
Thanks, @hoodmane. I'm just curious, what would be the difference between using `from js import eval` or `pyodide.run_js`?

**Comment by hoodmane** [comment]
Good question. The only difference is in the error handling. `eval(not_a_string)` returns `not_a_string` whereas `run_js` raises `ValueError`. But I kind of forgot we could just do `from js import eval` when I wrote this PR.

I think the better thing would be to provide `run_js` as a Python function rather than doing this stuff in C. I think raising a `ValueError` is better behavior when getting a weird argument than silently returning it. But then we could do:
```
def run_js(code, /):
	from js import eval

	if not isinstance(code, str):
		raise ValueError(f"...")
	eval(code)
```

File: src/py/pyodide/_run_js.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+def run_js(code, /):
```
Comment (rth) [inline src/py/pyodide/_run_js.py:1]: This needs a docstring.

**Review by rth — APPROVED** [review]
Indeed, it's a small improvement over JS eval, but still could be worth it,
 - to avoid overwriting python eval
 - to have better error checking
 - for better discoverability as part of the API docs

A minor comment otherwise LGTM.

**Pull request merged**